server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/http/AuthenticationRoutes.java

```diff
@@ -96,12 +101,13 @@ public Logger logger() {
     }
 
     @Override
-    public HttpServerRoutes define(HttpServerRoutes builder) {
-        return builder
-            .post(AUTHENTICATION, JMAPRoutes.corsHeaders(this::post))
-            .get(AUTHENTICATION, JMAPRoutes.corsHeaders(this::returnEndPointsResponse))
-            .delete(AUTHENTICATION, JMAPRoutes.corsHeaders(this::delete))
-            .options(AUTHENTICATION, CORS_CONTROL);
+    public Stream<JMAPRoute> routes() {
+        return Stream.of(
+            new JMAPRoute(new Endpoint(HttpMethod.POST, AUTHENTICATION), Version.DRAFT, JMAPRoutes.corsHeaders(this::post)),
```

Comment:
I'm still not happy with the repetition of the version.
What I tried to suggest was:
Have a top-level class per protocol version that get injected with routes and associate these routes to the version in this single place.

Also, I'd like JMAPRoute to offer a nice builder to avoid having to wrap manually methods with `corsHeaders`